### ISSUE TITLE
[BUG] Refrapt crashes in Test mode

### ISSUE THREAD

**Issue by ostruvek** [Progeny42/refrapt#25]
**Describe the bug**
When trying to test refrapt configuration, it crashes will following in the terminal:
[2022-06-07 12:27:13] INFO: Starting Refrapt process
[2022-06-07 12:27:13] INFO: ## Running in Test mode ##

[2022-06-07 12:27:13] INFO: The previous Refrapt run was interrupted. Full processing will be performed to ensure completeness
[2022-06-07 12:27:13] INFO: Removing previous /var files...

[2022-06-07 12:27:13] INFO: Processing 10 Repositories...
[2022-06-07 12:27:13] INFO: Compiled a list of 30 Release files for download
[2022-06-07 12:27:13] INFO: Downloading 30 Release files...
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████| 30/30 [00:00<00:00, 32.07 file/s]

[2022-06-07 12:27:15] INFO: Compiled a list of 560 Index files for download
[2022-06-07 12:27:15] INFO: Downloading 560 Index files...
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████| 560/560 [00:19<00:00, 28.07 file/s]

[2022-06-07 12:27:35] INFO: Decompressing Packages / Sources Indices...
Repositories : 100%|███████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:02<00:00,  4.20 repo/s]
                                                                                                                                                  
[2022-06-07 12:27:37] INFO: Building file list...
Traceback (most recent call last):                                                                                                                
  File "/usr/local/bin/refrapt", line 8, in <module>                                                                                              
    sys.exit(main())
  File "/home/stastny/.local/lib/python3.9/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/home/stastny/.local/lib/python3.9/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/home/stastny/.local/lib/python3.9/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/stastny/.local/lib/python3.9/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/home/stastny/.local/lib/python3.9/site-packages/refrapt/refrapt.py", line 120, in main
    PerformMirroring()
  File "/home/stastny/.local/lib/python3.9/site-packages/refrapt/refrapt.py", line 238, in PerformMirroring
    filesToDownload += repository.ParseIndexFiles() 
  File "/home/stastny/.local/lib/python3.9/site-packages/refrapt/classes.py", line 353, in ParseIndexFiles
    fileList += self._ProcessIndex(Settings.SkelPath(), file, False)
  File "/home/stastny/.local/lib/python3.9/site-packages/refrapt/classes.py", line 459, in _ProcessIndex
    packageList.append(Package(os.path.normpath(f"{path}/{directory}/{filename}"), int(package["Size"]), skipUpdateCheck or not self._NeedUpdate(os.path.normpath(f"{mirror}/{directory}/{filename}"), int(package["Size"]))))
KeyError: 'Size'

**Configuration File and Logs**
My config file:
```
##############
## SETTINGS ##
##############
# Each of the settings here are based on the default settings in Refrapt. If the values are
# not specified in this file, the default in Refrapt will be used instead.

# This is the default Architecture to be used if one is not specified in for a Repository.
set architecture       = amd64

# For each of the path variables, if not specified, will be found in ~/refrapt
# The root path for log files and the configuration file.
set rootPath   = "/etc/refrapt"
# The directory where the mirror will be stored. This will be the larget directory, so use a large drive!
set mirrorPath = "/var/www/html/ubuntu/mirror/"
# Directory used for intermediate Index files. Recommend using an SSD for best performance during the building of file lists.
set skelPath   = "/var/www/html/ubuntu/skel"
# Directory used for storing log files and lock files. Recommend using an SSD for best performance.
set varPath    = "/var/www/html/ubuntu/var"

# Specify whether the Contents-[arch].* files should be downloaded.
set contents           = True
# The number of threads to use for download and decompression. 
# If not specified here, Refrapt will get the number available from the machine.
set threads            = 6
# Specify whether to add the --auth-no-challenge parameter to Wget.
set authNoChallenge    = False
# Specify whether to add the --no-check-certificate parameter to Wget.
set noCheckCertificate = False
# Specify whether to add the --unlink parameter to Wget.
set unlink             = False
# Specify whether you are using a proxy for Wget.
set useProxy           = False
# The HTTP proxy to use. useProxy setting must be True.
set httpProxy          = ""
# The HTTPS proxy to use. useProxy setting must be True.
set httpsProxy         = ""
# The name of the proxy user. useProxy setting must be True.
set proxyUser          = ""
# The password of the proxy user. useProxy setting must be True.
set proxyPass          = ""
# The SSL certificate to pass to Wget.
set certificate        = ""
# The SSL certificate to pass to Wget.
set caCertificate      = ""
# The SSL key to pass to Wget.
set privateKey         = ""
# Set the --limit-rate value for each Wget instance. 
# Note that this will be multipled by the number of threads specified.
# See Wget manpage for correct syntax
set limitRate          = 500m # 500 MB
# Limits the download of Translation files for that of your locale. When not specified, Refrapt will default to locale.getdefaultlocale().
# Note that "_XX" languages will be stripped of the locale to collect a wider number of files. Ie "en_GB" becomes "en", which will capture 
# files with the same base language, but different locales.
# Multiple Languages can also be listed in a comma separated list; for example "en_GB, de_DE".
# set language           = "en_GB"
# Tell Refrapt to update all files, regardless of whether they are deemed to have changed or not.
# Note that if Wget determines the files are unchanged (via Timestamping), this will have no effect.
# The use of this should be limited for when you know a package has changed, but the size has not 
# (which is how Refrapt determines whether the file needs updating)
set forceUpdate        = False
# The log level to use for the application. Possible values are:
# DEBUG
# INFO
# WARNING
# ERROR
# CRITICAL
set logLevel           = DEBUG
# Will cause Refrapt to only process the Index files, and determine the list of Binary / Sources files to download, 
# and their size, but will not perform the main download. Useful for determining a potential download size before
# committing to it.
set test               = True
# Specify whether Refrapt should download the /by-hash/* directories and contents for each available checksum.
set byHash             = False
# Specify whether to disable Cleaning for all mirrors in this configuration file.
set disableClean       = False

#####################
## REPOSITORY LIST ##
#####################
# Define each of the Repositories which you wish to mirror.
# Multiple architectures can be specified per mirror, as well as multiple components

# Simple multi-component mirror
# deb [arch=amd64] http://gb.archive.ubuntu.com/ubuntu focal main restricted universe multiverse

# Multi-architecture mirror
# deb [arch=amd64,i386,armhf] https://repos.influxdata.com/debian buster stable

# Source mirror
# deb-src http://gb.archive.ubuntu.com/ubuntu focal main

# Flat repository
# deb http://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64

deb http://archive.ubuntu.com/ubuntu focal main restricted universe multiverse
deb http://archive.ubuntu.com/ubuntu focal-security main restricted universe multiverse
deb http://archive.ubuntu.com/ubuntu focal-updates main restricted universe multiverse
deb http://archive.ubuntu.com/ubuntu focal-proposed main restricted universe multiverse
deb http://archive.ubuntu.com/ubuntu focal-backports main restricted universe multiverse

deb-src http://archive.ubuntu.com/ubuntu focal main restricted universe multiverse
deb-src http://archive.ubuntu.com/ubuntu focal-security main restricted universe multiverse
deb-src http://archive.ubuntu.com/ubuntu focal-updates main restricted universe multiverse
deb-src http://archive.ubuntu.com/ubuntu focal-proposed main restricted universe multiverse
deb-src http://archive.ubuntu.com/ubuntu focal-backports main restricted universe multiverse

###################
## DISABLE CLEAN ##
###################
# Cleaning of unrequired files is enabled by default. To prevent cleaning 
# a particular directory, add the following command

# clean=False deb http://gb.archive.ubuntu.com/ubuntu
```
Log contains a lot of information, but nothing related to this issue. I haven't tried to run it without test mode.

**System Information:**
 - OS: Ubuntu 20.04.2 LTS
refrapt, version 0.4.2

**Comment by Progeny42** [comment]
A quick fix, just a small mistake in the code when getting the Size of the File when it's a deb-src repository. It will be available in v0.4.3 which I have just released.

To update perform the following:
```
python3.9 -m pip uninstall refrapt
python3.9 -m pip install refrapt
refrapt --version
```

The output should show that you have 0.4.3 installed.

**Comment by ostruvek** [comment]
Thank you for quick fix, it works like a charm right now!

`[2022-06-07 20:22:56] INFO: Starting Refrapt process
[2022-06-07 20:22:56] INFO: ## Running in Test mode ##

[2022-06-07 20:22:56] INFO: The previous Refrapt run was interrupted. Full processing will be performed to ensure completeness
[2022-06-07 20:22:56] INFO: Removing previous /var files...

[2022-06-07 20:22:56] INFO: Processing 10 Repositories...
[2022-06-07 20:22:56] INFO: Compiled a list of 30 Release files for download
[2022-06-07 20:22:56] INFO: Downloading 30 Release files...
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████| 30/30 [00:01<00:00, 19.70 file/s]

[2022-06-07 20:22:58] INFO: Compiled a list of 560 Index files for download
[2022-06-07 20:22:58] INFO: Downloading 560 Index files...
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████| 560/560 [00:38<00:00, 14.68 file/s]

[2022-06-07 20:23:36] INFO: Decompressing Packages / Sources Indices...
Repositories : 100%|███████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:02<00:00,  4.16 repo/s]
                                                                                                                                                      
[2022-06-07 20:23:38] INFO: Building file list...
[2022-06-07 20:23:54] INFO: Compiled a list of 200406 Binary and Source files of size 497.87 GB for download                                          
                                                                                                                                                      
[2022-06-07 20:23:54] INFO: Beginning Clean process...                                                                                                
[2022-06-07 20:23:54] INFO: 	Processing unmodified Indices...
[2022-06-07 20:23:59] INFO:     Compiling list of files to clean...                                                                                   
[2022-06-07 20:23:59] INFO:     No files eligible to clean                                                                                            

[2022-06-07 20:23:59] INFO: Test mode - Removing Release and Index files from /skel

[2022-06-07 20:23:59] INFO: Refrapt completed in 0:01:03`
